tokio-trace/tokio-trace-core/src/dispatcher.rs

```diff
@@ -63,6 +73,30 @@ pub fn with_default<T>(dispatcher: &Dispatch, f: impl FnOnce() -> T) -> T {
     let _guard = State::set_default(dispatcher.clone());
     f()
 }
+
+/// Sets this dispatch as the global default for the duration of the entire program.
+/// Will be used as a fallback if no thread-local dispatch has been set in a thread
+/// (using `with_default`.)
+///
+/// Can only be set once; subsequent attempts to set the global default will fail.
+/// Returns Err if the global default has already been set.
+///
+/// [span]: ../span/index.html
+/// [`Subscriber`]: ../subscriber/trait.Subscriber.html
+/// [`Event`]: ../event/struct.Event.html
+pub fn set_global_default(dispatcher: Dispatch) -> Result<(), ()> {
```

Comment:
This should probably return a type implementing `std::error::Error` --- something like 
```rust
pub struct SetDefaultError {
    _p: () // a private field prevents user code from constructing this type
}

impl fmt::Display for SetDefaultError {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        f.pad("a global default trace dispatcher has already been set")
    }
}

impl error::Error for SetDefaultError {}
```